### PR TITLE
[MultiKueue] Promote MultiKueue API and feature gate to Beta.

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

[MultiKueue] Promote MultiKueue API and feature gate to Beta.

The MultiKueue specific types are moved from `v1alpha1` API to `v1beta1`. 
The MultiKueue feature gate is promoted to beta.

#### Which issue(s) this PR fixes:

Related to #3095

#### Special notes for your reviewer:

This is marked only as `Related to #3095` so the issue is not closed before all it's prerequisites are. 

#### Does this PR introduce a user-facing change?

```release-note
Promote MultiKueue API and feature gate to Beta.

AR:
- The MultiKueue specific types are now part of the Kueue's `v1beta1` API.
- The MultiKueue feature gate is now beta and enabled by default.
```

### CODE REVIEW

**Review by mwielgus — COMMENTED** [review]
/lgtm

File: test/integration/webhook/jobs/suite_test.go [hunk 1]
```diff
@@ -71,6 +73,10 @@ var _ = ginkgo.AfterSuite(func() {
 
 func managerSetup(setup func(ctrl.Manager, ...jobframework.Option) error, opts ...jobframework.Option) framework.ManagerSetup {
 	return func(ctx context.Context, mgr manager.Manager) {
+		cCache := cache.New(mgr.GetClient())
+		queues := queue.NewManager(mgr.GetClient(), cCache)
+		opts = append(opts, jobframework.WithCache(cCache), jobframework.WithQueues(queues))
+
```
Comment (tenzen-y) [inline test/integration/webhook/jobs/suite_test.go:79]: What is the reason for the initialization of the cache and queue?

**Review by tenzen-y — COMMENTED** [review]
Otherwise, lgtm.

File: test/integration/webhook/jobs/suite_test.go (hunk 1 above)
Comment (trasc) [reply test/integration/webhook/jobs/suite_test.go:79]: Wit MK enabled (by default) the defaulting webhook is now hitting:

https://github.com/kubernetes-sigs/kueue/blob/6cace04a628195f52fad302e10127cdbbad6a380/pkg/controller/jobs/jobset/jobset_webhook.go#L72-L89

so we need to provide a cache and a queue manager.

Comment (mimowo) [reply test/integration/webhook/jobs/suite_test.go:79]: sgtm

**Comment by mimowo** [comment]
/approve
I believe the question https://github.com/kubernetes-sigs/kueue/pull/3230#discussion_r1800009063 is answered, and the other comment indicates lgtm https://github.com/kubernetes-sigs/kueue/pull/3230#pullrequestreview-2367500124, so I believe we can move it forward.

File: test/integration/webhook/jobs/suite_test.go (hunk 1 above)
Comment (tenzen-y) [reply test/integration/webhook/jobs/suite_test.go:79]: Ah, that makes sense.
Thank you for the clarifications!

**Comment by alculquicondor** [comment]
What does `AR` in the release notes stand for?

**Comment by tenzen-y** [comment]
> What does `AR` in the release notes stand for?

I guess `ActionRequired`.

**Comment by tenzen-y** [comment]
/release-note-edit

```release-note
Promote MultiKueue API and feature gate to Beta.

ACTION REQUIRED:
- The MultiKueue specific types are now part of the Kueue's `v1beta1` API.
- The MultiKueue feature gate is now beta and enabled by default.
```

**Comment by alculquicondor** [comment]
/release-note-edit
```release-note
Promote MultiKueue API and feature gate to Beta. The MultiKueue feature gate is now beta and enabled by default.

ACTION REQUIRED:
The MultiKueue specific types are now part of the Kueue's `v1beta1` API. `v1alpha` types are no longer supported.
```

**Pull request merged**